crypto/build.info

```diff
@@ -75,9 +75,12 @@ $UTIL_DEFINE=$CPUIDDEF
 
 SOURCE[../libcrypto]=$UTIL_COMMON \
         mem.c mem_sec.c \
-        cversion.c info.c cpt_err.c ebcdic.c uid.c o_time.c o_dir.c \
+        cversion.c info.c cpt_err.c uid.c o_time.c o_dir.c \
         o_fopen.c getenv.c o_init.c o_fips.c init.c trace.c provider.c \
         $UPLINKSRC
+IF[{- $config{options} ~= /-DCHARSET_EBCDIC/ -}]
+  SOURCE[../libcrypto]=ebcdic.c
+ENDIF
```

Comment:
I'm curious what other ways there are, but in order to not block progress I reverted the chane; fixup commit pushed.

Reply:
It might not come from configuration options, for example:

```
: ; git grep CHARSET_EBCDIC -- Configurations/
Configurations/10-main.conf:        cppflags         => "-DCHARSET_EBCDIC",
Configurations/50-os390.conf:        cflags           => "-O -DB_ENDIAN -DCHARSET_EBCDIC",
```

Reply:
yes, and both values end up in ```$confg{options}```  I looked.

Reply:
Another way is this, which I don't think is registered in `$config{options}`:

```
./config CPPFLAGS='-DCHARSET_EBCDIC'
```

or alternatively:

```
./config CPPDEFINES='CHARSET_EBCDIC'
```

Reply:
argh, I meant $config{cflags}   oh well.  one of those configs seems wrong to me.  shrug.

Reply:
> yes, and both values end up in ```$confg{options}```  I looked.

I sure would like to know how you checked.  The two config targets with those attributes are `BS2000-OSD` and `OS390-Unix`.

```
: ; ./Configure BS2000-OSD
Configuring OpenSSL version 3.0.0-dev for target BS2000-OSD
...
: ; grep '"options"' configdata.pm
    "options" => " no-afalgeng no-asan no-asm no-buildtest-c++ no-crypto-mdebug no-crypto-mdebug-backtrace no-devcryptoeng no-dynamic-engine no-ec_nistp_64_gcc_128 no-egd no-external-tests no-fips no-fuzz-afl no-fuzz-libfuzzer no-ktls no-md2 no-module no-msan no-pic no-rc5 no-sctp no-shared no-ssl-trace no-ssl3 no-ssl3-method no-threads no-trace no-ubsan no-unit-test no-uplink no-weak-ssl-ciphers no-zlib no-zlib-dynamic",
```

```
: ; ./Configure OS390-Unix
Configuring OpenSSL version 3.0.0-dev for target OS390-Unix
...
: ; grep '"options"' configdata.pm
    "options" => " no-afalgeng no-asan no-asm no-buildtest-c++ no-crypto-mdebug no-crypto-mdebug-backtrace no-devcryptoeng no-dynamic-engine no-ec_nistp_64_gcc_128 no-egd no-external-tests no-fips no-fuzz-afl no-fuzz-libfuzzer no-ktls no-md2 no-module no-msan no-pic no-rc5 no-sctp no-shared no-ssl-trace no-ssl3 no-ssl3-method no-threads no-trace no-ubsan no-unit-test no-uplink no-weak-ssl-ciphers no-zlib no-zlib-dynamic",
```

Nope, no `-DCHARSET_EBCDIC` in sight.

Reply:
As I said above, I was wrong; I meant the key was cflags.  But I don't care all that much and the change was reverted.

Reply:
> oh well. one of those configs seems wrong to me.

One is more up to date than the other, but they both actually work.  On Unix family operating systems, it doesn't *really* matter if `-D` are given as C preprocessor flags or C flags.  On other operating systems, it may matter.